pkg/controller/pcideviceclaim/pcideviceclaim_controller.go

```diff
@@ -133,20 +133,28 @@ func (h Handler) reconcilePCIDeviceClaims(hostname string) error {
 		// Check if PCI Device is already enabled for passthrough, but has no pre-existing PDC,
 		// if so, unbind the device (to force the user to make a proper PDC)
 		_, found := pdcNames[nodeAddr]
-		if !found && pd.Status.KernelDriverInUse == "vfio-pci" && hostname == pd.Status.NodeName {
+		if !found && pd.Status.KernelDriverInUse == "vfio-pci" && nodename == pd.Status.NodeName {
 			logrus.Infof("PCI Device %s is bound to vfio-pci but has no Claim, attempting to unbind", pd.Status.Address)
 			err = unbindPCIDeviceFromVfioPCIDriver(pd.Status.Address)
 			if err != nil {
 				return err
 			}
 		}
 		// After reboot, the PCIDeviceClaim will be there but the PCIDevice won't be bound to vfio-pci
-		if found && pd.Status.KernelDriverInUse != "vfio-pci" && hostname == pd.Status.NodeName {
+		if pd.Status.KernelDriverInUse != "vfio-pci" && nodename == pd.Status.NodeName {
 			// Set PassthroughEnabled to false
 			for _, pdc := range pdcs.Items {
 				if pdc.Spec.Address == pd.Status.Address {
 					logrus.Infof("Passthrough disabled for device %s", pd.Name)
 					pdc.Status.PassthroughEnabled = false
+					err = unbindPCIDeviceFromDriver(pd.Status.Address, pd.Status.KernelDriverInUse)
+					if err != nil {
```

Comment:
When `err != nil`, should return or not ?

if `unbindPCIDeviceFromDriver` failed but not return, then what will happen in `addNewIdToVfioPCIDriver` ?

Reply:
it will fail, we could `continue` here
sometimes there is no driver to unbind, as is, the function will fail, we should skip `unbindPCIDeviceFromDriver` in those cases

Reply:
That's right, I have worked with devices that didn't have a driver to unbind, we want to continue executing even if the unbind function fails

Reply:
With those exceptions, then it looks hard to identify the `devices that didn't have a driver to unbind`  and  `unbind failed`.

It is better to add comment here.  And the `logrus.Errorf ` needs also say like ` ..., the error is skipped`, maybe `logrus.Warnf` is the proper level.

Without those, pepole will have question to the code, the POD LOG will also bring surprise.